lib/package_provider/zip_file_generator.rb

```diff
@@ -0,0 +1,72 @@
+require 'zip'
+
+module PackageProvider
+  # This is a simple example which uses rubyzip to
+  # recursively generate a zip file from the contents of
+  # a specified directory. The directory itself is not
+  # included in the archive, rather just its contents.
+  #
+  # Usage:
```

Comment:
see yard or rdoc format:)
nice standards in API Documentation has rails: http://guides.rubyonrails.org/api_documentation_guidelines.html